### ISSUE TITLE
TTTTimeIntervalFormatter localizedIdiomaticDeicticExpressionForComponents incorrectly reports "last month", "last year", etc

### ISSUE THREAD

**Issue by eflath** [FormatterKit/FormatterKit#163]
The localized idiomatic deictic expression for dates spanning more than 1 month is still "last month". I worry that this is a fundamental failing of the logic--specifically the fact that we're attempting to construct a deictic expression for the difference between two dates based on a single NSDateComponents object.

Take, for example, the distance between March 5 and Jan 31. The components constructed between these dates (using `NSCalendar -components:fromDate:toDate:options:`) will be 

```
Calendar Year: 0
Month: -1
Day: -5
Hour: 0
Minute: 0
Second: 0
Week of Year: 0
```

the `month` value is -1, and so when passed through one of the localized `RelativeDateStringForComponents` methods, the result will be "Last Month". Obviously in March, we'd not refer to a date in January as "last month".

This can be extended to "Last Year" (take some date in early 2015 and compare with a date in late 2013) and even "Last Week" (a date fewer than 14 days ago that nevertheless spans across more than one calendar week)

As far as I can tell, it's impossible to construct an expression like this using a single DateComponents object. To fix this, I think we need to refactor `localizedIdiomaticDeicticExpression...` to pass both dates through, convert those both to components and take the difference between the relative component values.

**Comment by eflath** [comment]
Related bug: enter two dates that are exactly n weeks apart for n=1,2, or 3 with the more current date passed as the "from" date:

`[myFormatter stringForTimeIntervalFromDate:nowDate toDate:exactlyNWeeksInTheFuture]`

the result will be "just now"

move the endDate a minute forward, and the result will be "one minute from now" even though it is n weeks in the future.

**Comment by eflath** [comment]
*Update*: I've proposed a fix for this https://github.com/mattt/FormatterKit/pull/164
@ibsh has provided a fix to the related issue (which it seems we both discovered independently) https://github.com/mattt/FormatterKit/pull/162

**Comment by mattt** [comment]
Fixed by #189
